core/src/program/blocks/join_block.rs

```diff
@@ -12,6 +13,7 @@ use super::{fmt, hasher, Box, CodeBlock, Digest};
 pub struct Join {
     body: Box<[CodeBlock; 2]>,
     hash: Digest,
+    proc_info: Option<ProcInfo>,
```

Comment:
I think it can only happen if a procedure just calls another procedure (alias). I've changed it to have multiple markers in non-span blocks but if this is the only case, maybe not worth it.